src/sessions/softphone-session-handler.ts

```diff
@@ -66,42 +438,130 @@ export default class SoftphoneSessionHandler extends BaseSessionHandler {
       });
     }
 
+    /* make sure we store the session with the conversation state */
+    if (!this.conversations[session.conversationId]) {
+      this.conversations[session.conversationId] = { session } as any;
+    } else {
+      this.conversations[session.conversationId].session = session;
+    }
+
     await super.acceptSession(session, params);
     logDeviceChange(this.sdk, session, 'sessionStarted');
   }
 
-  async endSession (session: IExtendedMediaSession): Promise<void> {
+  async proceedWithSession (pendingSession: IPendingSession) {
+    if (!this.hasActiveSession() || pendingSession.id !== this.activeSession?.id) {
+      return super.proceedWithSession(pendingSession);
+    }
+
+    this.log('info', 'proceeding with proposed session via HTTP request', {
+      conversationId: pendingSession.conversationId,
+      sessionId: pendingSession.id
+    });
+
+    let participant = this.getUserParticipantFromConversationEvent(
+      this.conversations[pendingSession.conversationId]?.conversationUpdate
+    );
+
+    if (!participant) {
+      participant = await this.fetchUserParticipantFromConversationId(pendingSession.conversationId);
+    }
+
+    // const participant = await this.getParticipantForSession(pendingSession);
+    this.log('info', '`acceptSession` called with an active persistent connection. accepting via HTTP request', {
+      sessionId: pendingSession.id,
+      conversationId: pendingSession.conversationId
+    });
+
+    return this.patchPhoneCall(pendingSession.conversationId, participant.id, {
+      state: CommunicationStates.connected
+    });
+  }
+
+  async rejectPendingSession (pendingSession: IPendingSession): Promise<any> {
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=8384644)
```suggestion
  async rejectPendingSession (pendingSession: IPendingSession): Promise<unknown> {
```